tests/components/device/ace.spec.ts

```diff
@@ -556,10 +564,49 @@ describe('Ace', () => {
       }
     });
   });
+  describe('#getCnnFeatureStatus', () => {
+    sinon.spy(dummyTransport.grpcClient, 'getCnnFeatureStatus');
+    const cnnFeature = new huddly.CnnFeature();
+    cnnFeature.setFeature(huddly.Feature.AUTOZOOM);
+    it('should attempt to get a given cnn feature', async () => {
+      const cnnFeatureStatus = await device.getCnnFeatureStatus(cnnFeature);
+      const arg = dummyTransport.grpcClient.getCnnFeatureStatus.args[0][0];
+      expect(arg).to.be.instanceof(huddly.CnnFeature);
+      expect(arg.getFeature()).to.equal(huddly.Feature.AUTOZOOM);
+      expect(cnnFeatureStatus).to.be.instanceof(huddly.CNNStatus);
+    });
+    it('should handle error if something happens', async () => {
+      dummyTransport.grpcClient.getCnnFeatureStatus = (empty: Empty, cb: any) => {
+        cb(dummyError, undefined);
+      };
+      try {
+        await device.getCnnFeatureStatus(cnnFeature);
+      } catch (err) {
+        expect(err).to.equal(dummyError.message);
+      } finally {
+        expect(device.handleError).to.have.been.calledOnce;
+      }
+    });
+  });
+  describe('#getState', () => {
+    it('should return a object containing autozoom status (for the time being)', async () => {
+      sinon.stub(device, 'getCnnFeatureStatus').resolves(cnnStatusDummy);
+      const azStatus = await device.getState();
+      expect(azStatus).to.deep.equal({ autozoom_enabled: true });
+    });
+    it('should reject error if there is an issue', async () => {
+      sinon.stub(device, 'getCnnFeatureStatus').rejects('error');
+      try {
+        await device.getState();
```

Comment:
You can also do
```
const badFn = () => { device.getState(); };

return expect(badFn)..to.eventually.be.rejectedWith('Something went wrong');
```
Instead of doing try...catch blocks. 
Also nitpicking ;)

Reply:
Very nice pattern, and works well, but outputs 3 lines of warnings per test :S

Reply:
What kind of warning?

Reply:
`(node:3265) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)`

Reply:
Maybe you are calling it like this `return expect(badFn()).......`?